regression/invariants/driver.cpp

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************\
+
+Module: Invariant violation testing
```

Comment:
I think not - it isn't a unit test since it the test invovles reading from `stderr`. Therefore the test must be outside the executable. 

In fact I believe this is one of the main problems with the use of `abort()` rather than `throw` - it makes it impossible to unit test functions that should fail. 

To go back to @martin-cs's case for `abort` so no one can accidentally write a `catch(...)` and continue. I think it is more likely someone modifies the original function to skip over the invariant for some route even though it should still be true happen. A unit test that verifies the function still fails on incorrect input would catch that, but would be impossible to write without `throw`.